### PR TITLE
Add snapshot update yml, trigger snapshot update from pr label

### PR DESCRIPTION
**Pull request by PorterNan**
# What

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

**Review by alcail — APPROVED** [review]

**Review by alcail — APPROVED** [review]

**Comment by JamesBurnside** [comment]
> Failed to pass the UI Test. If this PR is for UI change and the error is snapshot mismatch, please add "ui change" label to the PR for updating the snapshot.

Good bot

File: .github/workflows/update-snapshot.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+      # Commit changes to snapshot files.
+      - name: Commit new snapshots
+        run: >
+          git diff --exit-code --quiet -- samples/StaticHtmlComposites/tests || 
+          (git add samples/StaticHtmlComposites/tests && 
+          git commit -m 'Updating automation snapshots' && 
+          git push origin HEAD:${{ github.head_ref }})
```
Comment (PorterNan) [inline .github/workflows/update-snapshot.yml:62]: @JamesBurnside Do you know how to reuse this part? Like create an external yml file and import it? This logic is reusable but not sure how to reuse it

**Comment by PorterNan** [comment]
We could actually catch some existing UI problems using this snapshot generate pipeline:

image

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -143,7 +143,8 @@ jobs:
         run: rush test -o chat
 
   build_static_html_composites_sample:
-    name: Build Static HTML Composites Sample
+    if: ${{ github.ref == 'refs/heads/main' || github.event_name == 'pull_request' }}
```
Comment (JamesBurnside) [inline .github/workflows/ci.yml:146]: The CI only runs on `push: [main]` and `pull_request` - is this if scoping this further?

File: .github/workflows/update-snapshot.yml (hunk 1 above)
Comment (JamesBurnside) [reply .github/workflows/update-snapshot.yml:62]: Not yet nicely that I've seen, its so sad 😔

File: .github/workflows/ci.yml (hunk 2 above)
Comment (PorterNan) [reply .github/workflows/ci.yml:146]: I believe I can remove this one, this is not needed anymore

File: .github/workflows/update-snapshot.yml [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          CONNECTION_STRING: ${{ secrets.CONNECTION_STRING }}
+      # Setup bot information for pushing new changes
+      # Here we use the id from the github actions bot: https://api.github.com/users/better-informatics%5Bbot%5D
+      - name: Setup bot git information
+        run: |
+          git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config user.name "github-actions[bot]"
+      # Commit changes to snapshot files.
+      - name: Commit new snapshots
+        run: >
```
Comment (JamesBurnside) [inline .github/workflows/update-snapshot.yml:58]: (question: what is `>` vs `|` in the unix command?)

File: docs/contributing-guide/6. submitting-a-pr.md [hunk 4]
```diff
@@ -43,6 +43,16 @@ It will ask you these questions for each package you have made changes in.
 
 Once that is done, commit these change files and include them in your PR.
 
+### Generating new snapshot for your UI changes
+If you are making an ui change in your PR, you might invalidate some screen snapshot in our automation visual tests. 
+
+1. Check the failing github actions to see if that is a desired UI change 
+   
+   ![module linking](../images/check-snapshots.png)
```
Comment (JamesBurnside) [inline docs/contributing-guide/6. submitting-a-pr.md:51]: nit: alt text should be more descriptive for visually impaired developers
```suggestion
   ![screenshot showing downloadable artifacts in a failed github action](../images/check-snapshots.png)
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -159,10 +159,73 @@ jobs:
         run: rush install
       - name: Build
         run: rush build -t sample-static-html-composites
-      - name: Tests
-        run: rush test -o sample-static-html-composites
+      - name: Automation Tests
```
Comment (JamesBurnside) [inline .github/workflows/ci.yml:162]: nit: can we name this task visual regression tests
```suggestion
      - name: Visual Regression Tests
```

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -159,10 +159,73 @@ jobs:
...
+    if: ${{ github.event_name == 'pull_request' && contains( github.event.pull_request.labels.*.name, 'ui change') }}
+    name: Build And Update Snapshot of Static HTML Composites Sample
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          # Use a machine account when checking out. This is to workaround the issue were GitHub
+          # actions, when using the default account, cannot trigger other actions.
+          # This machine account is only for this PAT, pwd was created and thrown away
```
Comment (JamesBurnside) [inline .github/workflows/ci.yml:197]: lol! 😆

**Review by JamesBurnside — APPROVED** [review]

File: .github/workflows/update-snapshot.yml (hunk 3 above)
Comment (PorterNan) [reply .github/workflows/update-snapshot.yml:58]: > indicates there are a single line command which will be separated into several lines
And | is for running multiple commands in following lines

Comment (JamesBurnside) [reply .github/workflows/update-snapshot.yml:58]: YouRockCatGIF

**Pull request merged**